### PR TITLE
Lynx 12.13.0 view history

### PR DESCRIPTION
**Pull request by hilak**

### CODE REVIEW

File: alpha/apps/kaltura/lib/kObjectDeleteHandler.php [hunk 1]
```diff
@@ -135,11 +135,20 @@ protected function entryDeleted(entry $entry)
 		
 		$c = new Criteria();
 		$c->add(categoryEntryPeer::ENTRY_ID, $entry->getId());
-		if(!categoryEntryPeer::doSelectOne($c)) {
+		if(categoryEntryPeer::doSelectOne($c)) {
+			kJobsManager::addDeleteJob($entry->getPartnerId(), DeleteObjectType::CATEGORY_ENTRY, $filter);
+		}
+		
+		$ueFilter = new UserEntryFilter();
```
Comment (MosheMaorKaltura) [inline alpha/apps/kaltura/lib/kObjectDeleteHandler.php]: cosmetic - var name ueFilter? rename to userEntryFilter

File: alpha/apps/kaltura/lib/kObjectDeleteHandler.php [hunk 2]
```diff
@@ -135,11 +135,20 @@ protected function entryDeleted(entry $entry)
...
+		$ueFilter->set("_eq_entry_id", $entry->getId());
+		
+		$c = new Criteria();
+		$c->add(UserEntryPeer::ENTRY_ID, $entry->getId());
+		if(!UserEntryPeer::doSelectOne($c)) {
 			return;
 		}
```
Comment (MosheMaorKaltura) [inline alpha/apps/kaltura/lib/kObjectDeleteHandler.php]: Move all the userEntryFilter related code into the brackets to work only if there is a record in the DB.

File: alpha/apps/kaltura/lib/kObjectDeleteHandler.php [hunk 3]
```diff
@@ -168,6 +177,17 @@ protected function kuserDelete(kuser $kuser)
...
 			}
 		}
+		
+		$ueFilter = new UserEntryFilter();
+		$ueFilter->set("_eq_user_id", $kuser->getPuserId());
+		
+		$c = new Criteria();
+		$c->add(UserEntryPeer::KUSER_ID, $kuser->getId());
+		if(!UserEntryPeer::doSelectOne($c)) {
+			return;
```
Comment (MosheMaorKaltura) [inline alpha/apps/kaltura/lib/kObjectDeleteHandler.php]: Same here
$userEntryFilter related code, to run only if there is on record of user entry in DB

File: alpha/lib/model/UserEntryPeer.php [hunk 4]
```diff
@@ -28,6 +28,7 @@ public static function setDefaultCriteriaFilter()
 	        self::$s_criteria_filter = new criteriaFilter();
 	    
 	    $c = KalturaCriteria::create(UserEntryPeer::OM_CLASS);
+		$c->addAnd ( UserEntryPeer::STATUS, array(UserEntryStatus::DELETED), Criteria::NOT_IN);
```
Comment (MosheMaorKaltura) [inline alpha/lib/model/UserEntryPeer.php]: Indetation

File: alpha/apps/kaltura/lib/kObjectDeleteHandler.php (hunk 2 above)
Comment (hilak) [inline alpha/apps/kaltura/lib/kObjectDeleteHandler.php]: we break out of the function entirely if there's not UserEntry record in the DB.

File: alpha/apps/kaltura/lib/kObjectDeleteHandler.php (hunk 3 above)
Comment (hilak) [inline alpha/apps/kaltura/lib/kObjectDeleteHandler.php]: we break out of the function entirely if there's not UserEntry record in the DB. The code after the if-statement will not run.

File: alpha/apps/kaltura/lib/kObjectDeleteHandler.php (hunk 2 above)
Comment (MosheMaorKaltura) [inline alpha/apps/kaltura/lib/kObjectDeleteHandler.php]: Meant something like this ->	
if(UserEntryPeer::doSelectOne($c)) {
		$ueFilter = new UserEntryFilter();
		$ueFilter->set("_eq_entry_id", $entry->getId());
		kJobsManager::addDeleteJob($entry->getPartnerId(), DeleteObjectType::USER_ENTRY, $ueFilter);
		}

**Pull request merged**